test/UniswapV3Staker.unit.spec.ts

```diff
@@ -397,14 +402,113 @@ describe('UniswapV3Staker.unit', async () => {
       })
     })
 
-    describe('via nft#safeTransferFrom', () => {
-      it('allows depositing without staking')
-      it('allows depositing and staking for a single incentive')
-      it('allows depositing and staking for two incentives')
-      it(
-        'reverts if staking information is less than 160 bytes and greater than 0 bytes'
-      )
-      it('reverts if staking information is invalid and greater than 160 bytes')
+    describe('nft#safeTransferFrom', () => {
+      let subject: (calldata: string, actor?: Wallet) => Promise<any>
+      let createIncentiveResult: HelperTypes.CreateIncentive.Result
+
+      beforeEach('setup', async () => {
+        const { startTime } = makeTimestamps(await blockTimestamp())
+
+        createIncentiveResult = await helpers.createIncentiveFlow({
+          rewardToken: context.rewardToken,
+          poolAddress: context.poolObj.address,
+          startTime,
+          totalReward,
+        })
+
+        await Time.setAndMine(startTime + 1)
+
+        // Make sure we're starting from a clean slate
+        const depositBefore = await context.staker.deposits(tokenId)
+        expect(depositBefore.owner).to.eq(constants.AddressZero)
+        expect(depositBefore.numberOfStakes).to.eq(0)
+
+        subject = (data: string, actor: Wallet = lpUser0) => {
+          return context.nft
+            .connect(actor)
+            [SAFE_TRANSFER_FROM_SIGNATURE](
+              actor.address,
+              context.staker.address,
+              tokenId,
+              data,
+              {
+                ...maxGas,
+                from: actor.address,
+              }
+            )
+        }
+      })
+
+      it('allows depositing without staking', async () => {
+        // Pass empty data
+        await subject(ethers.utils.defaultAbiCoder.encode([], []), lpUser0)
+        const deposit = await context.staker.deposits(tokenId)
+        expect(deposit.owner).to.eq(lpUser0.address)
+        // Make sure it is not staked.
+        expect(deposit.numberOfStakes).to.eq(BN('0'))
+      })
+
+      it('allows depositing and staking for a single incentive', async () => {
+        const data = ethers.utils.defaultAbiCoder.encode(
+          [INCENTIVE_KEY_ABI],
+          [incentiveResultToStakeAdapter(createIncentiveResult)]
+        )
+        await subject(data, lpUser0)
+        const deposit = await context.staker.deposits(tokenId)
+        expect(deposit.owner).to.eq(lpUser0.address)
+        expect(deposit.numberOfStakes).to.eq(BN('1'))
```

Comment:
should we check for the rest of the data as well? e.g. `incentive#numberOfStakes` and `stakes[tokenId][incentiveId]`